src/powershell/Program.cs

```diff
@@ -327,6 +327,12 @@ private static int ExecPwshLogin(string[] args, string pwshPath, bool isMacOS)
                 return Exec("/bin/zsh", execArgs);
```

Comment:
On macOS,`/bin/sh` is, in essence, `bash` (it's a separate binary that is a custom build of `bash` with a few changes hard-coded for better POSIX-compliance; verify with `sh --version`). As such, it _does_ support the `-l` option, and I believe it should be used - no need to bring `zsh` into the mix, which can have side effects.

Reply:
Thinking some more about this: Since `zsh` is being invoked in POSIX-compatibility mode, there's probably little practical difference to using `sh`.

The `zsh` executable is about double the size of `sh`, but in terms of execution _speed_ users probably won't notice the difference.

If `zsh` continues to be used, I suggest at least amending the misleading comment.

Reply:
No, we tried this. The version of bash it uses is too old to support `-l` and it wasn't starting properly. It's why the specific `/bin/zsh` behaviour exists.

The comment reflects the actual experimentation we did.

Anyway, this PR is about what shell we start with on Linux.

Reply:
Understood re focus of this PR; I just wanted to get the conversion started. 

`bash` on macOS has been frozen at `v3.2.57` for many version of macOS, for licensing reasons, and won't get updated.

Even if it gets removed, `/bin/sh` (the _system_ default, shell as opposed to the _interactive user_ default shell, which is now `zsh`) must and will remain a (mostly) POSIX-compliant shell - whatever concrete shell act as `/bin/sh`.

`sh` on macOS definitely supports the `-l` option - just try `/bin/sh -l`, and you'll see that it processes both `/etc/profile` and `~/.profile` without complaint (assuming POSIX-compliant scripts).

Are you saying that the option is supported but not _correctly_?

Reply:
I've given the tangent a new home: #12134

Also, with respect to _this_ PR: I suggest holding off until we understand the root cause of #12020